### PR TITLE
fix: make from_str parse 0x-prefixed strings

### PR DESCRIPTION
**Pull request by gakonst**
As title

Travis' error seems unrelated
image

### CODE REVIEW

File: uint/src/uint.rs [hunk 1]
```diff
@@ -1688,6 +1688,11 @@ macro_rules! construct_uint {
 			type Err = $crate::FromHexError;
 
 			fn from_str(value: &str) -> $crate::core_::result::Result<$name, Self::Err> {
+                let value = if value.starts_with("0x") {
```
Comment (dvdplm) [inline uint/src/uint.rs:1691]: Indentation looks off here.

File: fixed-hash/src/hash.rs [hunk 2]
```diff
@@ -588,6 +588,7 @@ macro_rules! impl_rustc_hex_for_fixed_hash {
 			/// - When encountering invalid non hex-digits
 			/// - Upon empty string input or invalid input length in general
 			fn from_str(input: &str) -> $crate::core_::result::Result<$name, $crate::rustc_hex::FromHexError> {
+				let input = if input.starts_with("0x") { &input[2..] } else { input };
```
Comment (dvdplm) [inline fixed-hash/src/hash.rs:591]: Prefer multiline here.

File: uint/tests/uint_tests.rs [hunk 3]
```diff
@@ -132,7 +132,9 @@ fn uint256_from() {
 	);
 
 	// test initializtion from string
+	let sa2 = U256::from_str("0x0a").unwrap();
 	let sa = U256::from_str("0a").unwrap();
+	assert_eq!(sa2, sa);
```
Comment (dvdplm) [inline uint/tests/uint_tests.rs:137]:
```suggestion
	let sa = U256::from_str("0a").unwrap();
	let sa2 = U256::from_str("0x0a").unwrap();
	assert_eq!(sa2, sa);
```

**Review by dvdplm — CHANGES_REQUESTED** [review]

File: uint/src/uint.rs (hunk 1 above)
Comment (gakonst) [reply uint/src/uint.rs:1691]: Seems like a github bug. It doesn't show in the Full review tab, and cargo fmt doesn't catch it

image

image

Comment (dvdplm) [reply uint/src/uint.rs:1691]: It looks like you're using spaces (all crates here use tabs, `\t`) and that's why it renders weird on github. Can you convert to tabs please?

**Comment by gakonst** [comment]
@dvdplm I've addressed your changes, would appreciate another look - thanks!

**Review by dvdplm — APPROVED** [review]

**Comment by gakonst** [comment]
@niklasad1 @ordian would be great if you could approve this PR

**Review by ordian — APPROVED** [review]

**Comment by dvdplm** [comment]
@gakonst We need a `cargo fmt` I'm afraid.

**Comment by gakonst** [comment]
@dvdplm I did that initially and you said you prefer multiline :) Addressed in latest commit. Thanks!

File: fixed-hash/src/hash.rs (hunk 2 above)
Comment (ordian) [inline fixed-hash/src/hash.rs:591]: alternatively we could use
```suggestion
				let input = input.strip_prefix("0x").unwrap_or(input);
```

**Comment by dvdplm** [comment]
> @dvdplm I did that initially and you said you prefer multiline :) Addressed in latest commit. Thanks!

I know, my bad. Apologies. :(

**Comment by gakonst** [comment]
Switched to use `strip_prefix`, good rec! PR should be good to merge now.

File: uint/src/uint.rs [hunk 4]
```diff
@@ -1688,6 +1688,7 @@ macro_rules! construct_uint {
 			type Err = $crate::FromHexError;
 
 			fn from_str(value: &str) -> $crate::core_::result::Result<$name, Self::Err> {
+                let value = value.strip_prefix("0x").unwrap_or(value);
```
Comment (ordian) [inline uint/src/uint.rs:1691]: illegal spaces!

**Pull request merged**